service/src/main/java/bio/terra/catalog/service/dataset/DatasetDao.java

```diff
@@ -96,25 +96,26 @@ public Dataset update(Dataset dataset) {
             + "WHERE id = :id";
     MapSqlParameterSource params =
         new MapSqlParameterSource()
-            .addValue(ID_FIELD, dataset.id())
+            .addValue(ID_FIELD, dataset.id().uuid())
             .addValue(STORAGE_SOURCE_ID_FIELD, dataset.storageSourceId())
             .addValue(STORAGE_SYSTEM_FIELD, String.valueOf(dataset.storageSystem()))
             .addValue(METADATA_FIELD, dataset.metadata());
     return createOrUpdate(sql, params);
   }
 
   @WriteTransaction
-  public boolean delete(UUID id) {
+  public boolean delete(Dataset dataset) {
```

Comment:
Why change this to `dataset` here if we only need the id?

Reply:
I changed this because the caller always has a `Dataset` on hand before calling this method, because the dataset's storage system is required to perform the permission check. So it seemed more consistent to accept `Dataset` here. But you're right, I could use accept `DatasetId` instead and have the caller extract the ID from the dataset object instead.